src/library/dlgtagfetcher.cpp

```diff
@@ -402,4 +521,162 @@ void DlgTagFetcher::tagSelected() {
+    int fetchedCoverArtQualityConfigValue =
+            m_pConfig->getValue(mixxx::library::prefs::kCoverArtFetcherQualityConfigKey,
+                    static_cast<int>(DlgPrefLibrary::CoverArtFetcherQuality::Low));
+
+    DlgPrefLibrary::CoverArtFetcherQuality fetcherQuality =
+            static_cast<DlgPrefLibrary::CoverArtFetcherQuality>(
+                    fetchedCoverArtQualityConfigValue);
+
+    // Cover art links task can retrieve us variable number of links with different cover art sizes
+    // Every single successful response has 2 links.
+    // These links are cover arts with 250 PX and 500 PX
+    // Some of the tags might have link of a cover art with 1200 PX
+    // Some of the tags might have link of a cover art more than 1200 PX
+    // At final, we always retrieve 2 links, but for some tags this can be 3 or 4
+    // We need to pick the correct size according to user choice
+
+    // User choices and the possible fetched cover art sizes are:
+    // Highest -> More than 1200 PX, if not available 1200 PX or 500 PX
+    // High    -> 1200 PX if not available 500 PX
+    // Medium  -> Always 500 PX fetched
+    // Low     -> Always 250 PX fetched
+
+    if (fetcherQuality == DlgPrefLibrary::CoverArtFetcherQuality::Highest) {
+        getCoverArt(allUrls.last());
+        return;
+    } else if (fetcherQuality == DlgPrefLibrary::CoverArtFetcherQuality::High) {
+        allUrls.size() < 3 ? getCoverArt(allUrls.last()) : getCoverArt(allUrls.at(2));
+        return;
+    } else if (fetcherQuality == DlgPrefLibrary::CoverArtFetcherQuality::Medium) {
+        getCoverArt(allUrls.at(1));
+        return;
+    } else {
+        getCoverArt(allUrls.first());
+    }
+}
+
+void DlgTagFetcher::slotLoadBytesToLabel(const QByteArray& data) {
+    QPixmap fetchedCoverArtPixmap;
+    fetchedCoverArtPixmap.loadFromData(data);
+    CoverInfo coverInfo;
+    coverInfo.type = CoverInfo::NONE;
+    coverInfo.source = CoverInfo::USER_SELECTED;
+    coverInfo.setImage();
+
+    loadingProgressBar->setVisible(false);
+    QString coverArtMessage = tr("Cover art is ready to be applied");
+    statusMessage->setVisible(true);
+    statusMessage->setText(coverArtMessage);
+
+    m_fetchedCoverArtByteArrays = data;
+    m_pWFetchedCoverArtLabel->loadData(
+            m_fetchedCoverArtByteArrays); //This data loaded because for full size.
+    m_pWFetchedCoverArtLabel->setCoverArt(coverInfo, fetchedCoverArtPixmap);
+}
+
+void DlgTagFetcher::getCoverArt(const QString& url) {
+    QString coverArtMessage = tr("Cover art found getting image");
+    loadingProgressBar->setFormat(coverArtMessage);
+    loadingProgressBar->setValue(kPercentForCoverArtImageTask);
+    m_tagFetcher.startFetchCoverArtImage(url);
+}
+
+void DlgTagFetcher::slotCoverArtLinkNotFound() {
+    loadingProgressBar->setVisible(false);
+    statusMessage->setVisible(true);
+    QString message = tr("Cover Art is not available for selected tag");
```

Comment:
```suggestion
    QString message = tr("Cover Art is not available for selected row");
```
or metadata, or release